EpsilonAuxBMS/Tests/tests/UpdateChargeAllowanceTaskTest/UpdateChargeAllowanceTaskTest.c

```diff
@@ -0,0 +1,178 @@
+{
+    auxStatus.startUpSequenceDone = 1;
+    HAL_GPIO_ReadPin_ExpectAndReturn(ORION_DISCHARGE_ENABLE_SENSE_GPIO_Port, ORION_DISCHARGE_ENABLE_SENSE_Pin, expectedOrionDischargeSense);
+    HAL_GPIO_ReadPin_ExpectAndReturn(ORION_CHARGE_ENABLE_SENSE_GPIO_Port, ORION_CHARGE_ENABLE_SENSE_Pin, expectedOrionChargeSense);
+}
+
+void test_UpdateChargeAllowanceTask_StandardOperationShouldUpdateAuxAndOrionStatusCorrectly()
+{
+    uint32_t prevWakeTime = 0;
+    setUpdateChargeAllowanceOsExpects(&prevWakeTime);
+    GPIO_PinState expectedOrionDischargeSense = GPIO_PIN_SET;
+    GPIO_PinState expectedOrionChargeSense = GPIO_PIN_SET;
+
+    setOrionEnableSenseExpects(expectedOrionDischargeSense, expectedOrionChargeSense);
+
+    carShouldTrip_ExpectAndReturn(0);
+
+    orionStatus.canMsgReceived = 1;
+    orionStatus.maxCellVoltage = 30000;
+    orionStatus.minCellVoltage = 30000;
+    updateChargeAllowance(&prevWakeTime);
+
+    TEST_ASSERT_EQUAL(1, orionStatus.allowDischarge);
+    TEST_ASSERT_EQUAL(1, orionStatus.allowCharge);
+    TEST_ASSERT_EQUAL(1, orionStatus.batteryVoltagesInRange);
+    TEST_ASSERT_EQUAL(0, orionStatus.shutOff);
+    TEST_ASSERT_EQUAL(0, orionStatus.chargeContactorOverriden);
+    TEST_ASSERT_EQUAL(0, orionStatus.dischargeContactorOverriden);
+
+    TEST_ASSERT_EQUAL(0, auxStatus.strobeBmsLight);
+    TEST_ASSERT_EQUAL(1, auxStatus.allowCharge);
+}
+
+void test_UpdateChargeAllowanceTask_NoDischargeSenseShouldTurnOffPinAndUpdateOrionStatus()
+{
+    uint32_t prevWakeTime = 0;
+    setUpdateChargeAllowanceOsExpects(&prevWakeTime);
+
+    GPIO_PinState expectedOrionDischargeSense = GPIO_PIN_RESET;
+    GPIO_PinState expectedOrionChargeSense = GPIO_PIN_SET;
+    setOrionEnableSenseExpects(expectedOrionDischargeSense, expectedOrionChargeSense);
+
+    carShouldTrip_ExpectAndReturn(0);
+
+    // Test if PIN discharge enable and HV was written to
+    HAL_GPIO_WritePin_Expect(HV_ENABLE_GPIO_Port, HV_ENABLE_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin_Expect(DISCHARGE_ENABLE_GPIO_Port, DISCHARGE_ENABLE_Pin, GPIO_PIN_RESET);
+    updateChargeAllowance(&prevWakeTime);
+
+    TEST_ASSERT_EQUAL(0, orionStatus.allowDischarge);
+    TEST_ASSERT_EQUAL(1, orionStatus.dischargeContactorOverriden);
+}
+
+void test_UpdateChargeAllowanceTask_NoChargeSenseShouldTurnOffPinAndUpdateAuxAndOrionStatus()
+{
+    uint32_t prevWakeTime = 0;
+    setUpdateChargeAllowanceOsExpects(&prevWakeTime);
+
+    GPIO_PinState expectedOrionDischargeSense = GPIO_PIN_SET;
+    GPIO_PinState expectedOrionChargeSense = GPIO_PIN_RESET;
+    setOrionEnableSenseExpects(expectedOrionDischargeSense, expectedOrionChargeSense);
+
+    carShouldTrip_ExpectAndReturn(0);
+
+    HAL_GPIO_WritePin_Expect(CHARGE_ENABLE_GPIO_Port, CHARGE_ENABLE_Pin, GPIO_PIN_RESET);
+    updateChargeAllowance(&prevWakeTime);
+
+    TEST_ASSERT_EQUAL(0, orionStatus.allowCharge);
+    TEST_ASSERT_EQUAL(0, auxStatus.allowCharge);
+    TEST_ASSERT_EQUAL(1, orionStatus.chargeContactorOverriden);
```

Comment:
Should probably check that the discharge related members are what is expected as well. Maybe turn into functions that checks for normal charge and discharge values.